### PR TITLE
lesson 3

### PR DESCRIPTION
**Pull request by netscro**
HW 3

### CODE REVIEW

File: src/com/javacore/lesson3/HomeWork.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.javacore.lesson3;
+
+
+public class HomeWork {
+
+    public static void main(String[] args) {
+
+        SuperArrayList<String> superArrayList = new SuperArrayList<>();
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/HomeWork.java]: As you have interface for your implementation better will be:
`BestList <String> superArrayList = new SuperArrayList<>();`

File: src/com/javacore/lesson3/SuperArrayList.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+public class SuperArrayList<Type> implements BestList<Type> {
+
+    private static final int LIST_SIZE = 0;
+    private static Object[] array = new Object[LIST_SIZE];
+    private static Object[] temporaryArray;
+
+    @Override
+    public Type get(int index) {
+        return (Type) array[index];
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/SuperArrayList.java:14]: What if no element by index?
For example, you pass index 10 and array length 5. Need to add validation here.

File: src/com/javacore/lesson3/SuperArrayList.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public void set(int index, Type value) {
+        array[index] = value;
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/SuperArrayList.java:20]: index validation

File: src/com/javacore/lesson3/SuperArrayList.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public void remove(int index) {
+        if (isNotEmptyArray()) {
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/SuperArrayList.java:26]: index validation

File: src/com/javacore/lesson3/SuperArrayList.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                }
+            }
+            array = temporaryArray;
+        }
+    }
+
+    @Override
+    public void add(int index, Type value) {
+
+        temporaryArray = new Object[size() + 1];
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/SuperArrayList.java:41]: index validation

**Review by kolodgeeva — CHANGES_REQUESTED** [review]

File: src/com/javacore/lesson3/SuperArrayList.java [hunk 6]
```diff
@@ -69,4 +72,12 @@ private boolean isNotEmptyArray() {
         }
         return true;
     }
+
+    private void indexValidation(int index){
+        if (index > size()){
```
Comment (kolodgeeva) [inline src/com/javacore/lesson3/SuperArrayList.java]: pls check negative values as well

**Comment by netscro** [comment]
Thanks!

**Pull request merged**